### PR TITLE
[SPARK-6975][Yarn] Fix argument validation error

### PR DESCRIPTION
**Pull request by jerryshao**
`numExecutors` checking is failed when dynamic allocation is enabled with default configuration. Details can be seen is [SPARK-6975](https://issues.apache.org/jira/browse/SPARK-6975). @sryza, please help me to review this, not sure is this the correct way, I think previous you change this part :)

### CODE REVIEW

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala [hunk 1]
```diff
@@ -103,9 +103,14 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
    * This is intended to be called only after the provided arguments have been parsed.
    */
   private def validateArgs(): Unit = {
-    if (numExecutors <= 0) {
+    if (numExecutors < 0 || (!isDynamicAllocationEnabled && numExecutors == 0)) {
       throw new IllegalArgumentException(
-        "You must specify at least 1 executor!\n" + getUsageMessage())
+        s"""
+           |Number of executors $numExecutors is not legal.
+           |If dynamic allocation is enable, number of executors should at least be 0.
```
Comment (srowen) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala]: enabled -> enabled. I think this is simpler to state as "Number of executors was $numExecutors, but must be at least 1 (or 0 if dynamic executor allocation is enabled)."

Comment (jerryshao) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala]: OK, I will change the statement.

**Comment by srowen** [comment]
Jenkins, retest this please.

**Comment by vanzin** [comment]
Looks fine to me. Seems like it might be useful (and easy) to have different messages for each case, but not a big deal.

**Comment by andrewor14** [comment]
This looks fine. The previous setting was very confusing to the user because they would have run into this by following the docs. Thanks for fixing this I'm merging this into master and 1.3

**Pull request closed without merging**